src/QA/QA.Web/Orchestrators/WithdrawVacancyOrchestrator.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Esfa.Recruit.Qa.Web.Models.WithdrawVacancy;
+using Esfa.Recruit.Qa.Web.ViewModels.WithdrawVacancy;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Domain.Extensions;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Exceptions;
+
+namespace Esfa.Recruit.Qa.Web.Orchestrators
+{
+    public class WithdrawVacancyOrchestrator
+    {
+        private readonly IQaVacancyClient _vacancyClient;
+
+        public WithdrawVacancyOrchestrator(IQaVacancyClient vacancyClient)
+        {
+            _vacancyClient = vacancyClient;
+        }
+
+        public FindVacancyViewModel GetFindVacancyViewModel()
+        {
+            return new FindVacancyViewModel();
+        }
+
+        public async Task<PostFindVacancyEditModelResult> PostFindVacancyEditModelAsync(FindVacancyEditModel m)
+        {
+            var vacancy = await GetVacancyAsync(m.VacancyReference.ToUpper().Replace("VAC", ""));
+
+            if(vacancy != null && vacancy.CanClose)
+                return new PostFindVacancyEditModelResult { ResultType = PostFindVacancyEditModelResultType.CanClose, VacancyReference = vacancy.VacancyReference };
+            
+            if(vacancy != null && vacancy.Status == VacancyStatus.Closed)
+                return new PostFindVacancyEditModelResult { ResultType = PostFindVacancyEditModelResultType.AlreadyClosed, VacancyReference = vacancy.VacancyReference };
+
+            return new PostFindVacancyEditModelResult {ResultType = PostFindVacancyEditModelResultType.NotFound};
+        }
+
+        public FindVacancyViewModel GetFindVacancyViewModel(FindVacancyEditModel m)
+        {
+            var vm = GetFindVacancyViewModel();
+
+            vm.VacancyReference = m.VacancyReference;
+
+            return vm;
+        }
+
+        public async Task<AlreadyClosedViewModel> GetAlreadyClosedViewModelAsync(string vacancyReference)
+        {
+            var vacancy = await GetVacancyAsync(vacancyReference);
+
+            if (vacancy == null || vacancy.CanClose)
+                return null;
+
+            return new AlreadyClosedViewModel
```

Comment:
Consider a parameterised constructor, so when new properties are added in future the compiler will automatically find every occurrence we need to update.